momepy/functional/_intensity.py

```diff
@@ -111,3 +113,43 @@ def _calc_nodedensity(group, edges):
         summation_values = pd.Series(np.ones(nodes.shape[0]), index=nodes.index)
 
     return graph.apply(summation_values, _calc_nodedensity, edges=edges)
+
+
+def count_unique(aggregation_key: Series, graph: Graph, areas: Series = None) -> Series:
+    """Calculates the (area weighted) number of unique \\
+       ``aggregation_key`` aggregations reachable in ``graph``.
+
+    The unique number of aggregations within neighbours defined in ``graph``,
+    optionally divided by the area covered by the neighbours.
+
+    Adapted from :cite:`dibble2017`.
+
+    Parameters
+    ----------
+    aggregation_key: pd.Series
+        The group key that denotes group membership of ``graph`` elements.
+    graph : libpysal.graph.Graph
+        A spatial weights matrix for the tessellations.
+    areas : Series, default None
+        Areas of the tessellations, if areas is none return unique count.
+
+    Returns
+    -------
+    Series
+
+
+    Examples
+    --------
+    >>> tessellation_df['blocks_within_4'] = mm.block_counts(
+    ...                                       df_tessellation['bID'],
+    ...                                       graph_queen4,
+    ...                                       buildings_df['uID'])
+    """
+
+    results = describe(aggregation_key, graph, include_nunique=True)["nunique"]
```

Comment:
the overhead of computing the rest is negligible here? or shall we make the rest optional as well so we compute just `nunique`?

Reply:
i think so, running the compute_stats function takes around 1 sec for around 100k buildings